packages/calcite-components/src/utils/filter.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { filter } from "./filter";
+
+describe("filter function", () => {
+  it("warns and returns empty array for empty data", () => {
+    const originalWarn = console.warn;
+    console.warn = jest.fn();
```

Comment:
Can you stub with jest instead of replacing it directly?

```tsx
jest.spyOn(console, "warn");
```

I think you'll also need to call `jest.restoreMocks` in an `afterEach` to restore the original `warn`.